### PR TITLE
Adding GitHub headers

### LINKED ISSUE #13580: [FEATURE] Include additional Github headers for webhook events
For all Github webhook triggers, at the same level as `action`, we should add a new object called `github_headers` - this will be useful for tracking, as well as debugging purposes.

- github_headers
  - `X-GitHub-Event`
  - `X-Github-Delivery`
  - `X-GitHub-Hook-ID`

### PR DESCRIPTION
**Pull request by GTFalcao**
Closes #13580 

## Summary by CodeRabbit

- **New Features**
	- Enhanced webhook handling with additional metadata from headers for improved event processing.
	- Introduced a utility function to extract relevant GitHub headers for better event context.

- **Bug Fixes**
	- Minor version updates across multiple components indicating potential bug fixes and enhancements.

### CODE REVIEW

**Review by lcaresia — APPROVED** [review]

**Review by michelle0927 — APPROVED** [review]

**Review by lcaresia — DISMISSED** [review]

**Comment by GTFalcao** [comment]
This has a conflict in an unrelated app file, likely from merging with master before - I'll need a new approval

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**
